helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/ClusterAccessor.java

```diff
@@ -321,8 +322,13 @@ public Response updateCluster(@PathParam("clusterId") String clusterId,
           customFieldsMap =
               OBJECT_MAPPER.readValue(content, new TypeReference<HashMap<String, String>>() {
               });
-          // content is given as a KV mapping. Nullify content
+          // content is given as a KV mapping. Nullify content unless (case-insensitive) reason key present in map
           content = null;
+          for (Map.Entry<String, String> entry : customFieldsMap.entrySet()) {
+            if ("reason".equals(entry.getKey().toLowerCase(Locale.ENGLISH))) {
```

Comment:
I think Locale.ENGLISH is not needed, right? We are not checking semantics or not.

If the key does not match, it is not right then.

Reply:
Habit from old team. I've updated to use equalsIgnoreCase:

if ("reason".**equalsIgnoreCase**(entry.getKey())) {